inteceptors/Server.cfc

```diff
@@ -8,39 +8,74 @@ component {
 		
 		if( !interceptData.settings.serverEnable ) { return; }
 		
-		// Map the server statuses to a color
-		var statusColors = {
-			running : 'greenOnWhite',
-			starting : 'yellowOnWhite',
-			stopped : 'redOnWhite'
-		};
+		var rootPath = fileSystem.resolvePath( '' );
+		var traversePath = '';
+		var carSettings = {};
+		var serverDetails = {};
+		
 		try {
-			var serverDetails = serverService.resolveServerDetails( directory = fileSystem.resolvePath( '' ), serverProps={} );
+
+			serverDetails = serverService.resolveServerDetails( directory = rootPath, serverProps={} );
+
 			if( !serverDetails.serverIsNew ) {
-				var serverInfo = serverDetails.serverInfo;
-				var status = serverService.isServerRunning( serverInfo ) ? 'running' : 'stopped';
+
+				carSettings = generateCar(serverDetails, interceptData);	
+				interceptData.cars.server.text = carSettings.text;
+				interceptData.cars.server.background = carSettings.background;
 				
-				if( serverInfo.WARPath.len() ) {
-					interceptData.cars.server.text = print.text( ' ' & serverInfo.WARPath.listLast( '/\' ) & ' '& '(', '#interceptData.settings.serverText#on#interceptData.settings.serverBG#' ) 
-						& print.text( status, statusColors[ status ] & 'on#interceptData.settings.serverBG#' ) 
-						& print.text( ') ', '#interceptData.settings.serverText#on#interceptData.settings.serverBG#' );
-				} else {
-					var version = semanticVersion.parseVersion( serverInfo.engineVersion );
-					interceptData.cars.server.text = print.text( ' ' & serverInfo.engineName & ' ' & version.major & '.' & version.minor & '.' & version.revision & ' '& '(', '#interceptData.settings.serverText#on#interceptData.settings.serverBG#' ) 
-						& print.text( status, statusColors[ status ] & 'on#interceptData.settings.serverBG#' ) 
-						& print.text( ') ', '#interceptData.settings.serverText#on#interceptData.settings.serverBG#' );	
+			}else{
+				for(var i = 1; i <= interceptData.settings.serverDepth; i++){
+					serverDetails = serverService.resolveServerDetails( serverProps={directory=rootPath} );
```

Comment:
Now notice when traversing up, the resolveServerDetails is called with a serverProps.directory instead like  its done in line 18. If I called it using the unrequired directory argument like in line 18, then ServerService would have to be modified.